service/instance.py

```diff
@@ -1340,22 +1341,105 @@ def user_delete_security_group(core_identity):
 
 def security_group_init(core_identity, max_attempts=3):
     has_secret = core_identity.get_credential('secret') is not None
+    security_group_name = core_identityprovider.get_config("network", "security_group_name", "atmosphere")
     if has_secret:
         return admin_security_group_init(core_identity)
-    return user_security_group_init(core_identity)
+    return user_security_group_init(core_identity, security_group_name = security_group_name)
 
 
-def user_security_group_init(core_identity, security_group_name='default'):
-    # Rules can come from the provider _or_ from settings _otherwise_ empty-list
-    rules = core_identity.provider.get_config('network', 'default_security_rules',getattr(settings,'DEFAULT_RULES',[]))
-    driver = get_cached_driver(identity=core_identity)
-    lc_driver = driver._connection
-    security_group = get_or_create_security_group(lc_driver, security_group_name)
-    set_security_group_rules(lc_driver, security_group, rules)
+def user_security_group_init(core_identity, security_group_name):
+    network_driver = _to_network_driver(core_identity)
+    user_neutron = network_driver.neutron
+    extended_default_rules = _get_default_rules()
+    user_security_group_rules = core_identity.provider.get_config('network', 'user_security_rules', extended_default_rules)
```

Comment:
For now, you would make this change in the python REPL and/or the `/admin` section for the specific provider in question.

_In a BYOC future,_ this will be handled in a cleaner admin section within troposphere that has well-defined segments and keys.